### DISCUSSION TITLE
Can you sign a NativeAOT module with a strong name?

### DISCUSSION THREAD

**Discussion by slxdy** [dotnet/runtime#94953] [Q&A] [answered]
I do not have a key yet, but I'd like to ask this before getting one.

**Comment by jkotas** [comment, marked as answer]
Strong name signing is specific to managed IL file format. Published native AOT binaries do not use the IL file format, so it is not possible to sign them with a strong name.

**Reply by slxdy** [reply]
Well, how would I sign my executable after NativeAOT then?

**Reply by MichalStrehovsky** [reply]
On Windows: https://learn.microsoft.com/en-us/windows/win32/appxpkg/how-to-sign-a-package-using-signtool
On Mac: https://developer.apple.com/library/archive/documentation/Security/Conceptual/CodeSigningGuide/Procedures/Procedures.html

I don't know if there is anything like that on Linux.

**Comment by hez2010** [comment]
Instead of signing the assembly, in NativeAOT what you need to do is to sign the executable file.

**Reply by jkotas** [reply]
Yes - it depends on what you are trying to achieve by the signing.

**Reply by MichalStrehovsky** [reply]
If it's Authenticode signing, that one should work. We ship several binaries compiled with Native AOT in .NET 8 that are Authenticode signed.
